### PR TITLE
Fix core notifications

### PR DESCRIPTION
**Pull request by MrMysteryMeat**
# Description

As Fuji described in https://github.com/TempleDAO/temple/issues/459, notifications should be fixed, so that you don't have to scroll up to see them, and the notification manager should handle closing notifications. I removed a lot of the complex logic in Notifications.tsx and changed it so that NotificationManager.tsx handles the timing of notifications. The code should be much more clear, I also added comments.

Fixes # (issue)

# Checklist
- [X] Code follows the style guide
- [X] I have performed a self-review of my own code
- [X] New and existing tests pass locally
- [X] This PR is targeting the correct branch

### CODE REVIEW

**Comment by MrMysteryMeat** [comment]
BY THE WAY: in staging, all notifications opened within the ten second time frame of the first notification will close once the timeframe of the first notification ends (with the same hash). Meaning that if the user gets a notification and then gets a notification nine seconds later, the second notification will only show for one second. In this version, all notifications will close 10 seconds after the last notification thrown, regardless of the hash

File: apps/dapp/src/components/Notification/NotificationManager.tsx [hunk 1]
```diff
@@ -1,16 +1,54 @@
...
+
+const AUTOHIDE_TIMER = 1000 * 10;
 
 const NotificationManager = () => {
   const { notifications } = useNotification();
 
+  const { closeNotification } = useNotification();
+
+  // Array indicating which notifications are open.
+  // I know what you're thinking - "couldn't this just
```
Comment (nichosystem) [inline apps/dapp/src/components/Notification/NotificationManager.tsx:14]: appreciate the context, we can cut this extended comment for now 🙂

File: apps/dapp/src/components/Notification/NotificationManager.tsx [hunk 2]
```diff
@@ -1,16 +1,54 @@
...
+  // be a boolean because we are always setting the whole
+  // array to either true of false?" Well I tested that,
+  // and it messes up the animation when opening multiple
+  // notifications at once, so let's keep it like this.
+  const [areOpen, setAreOpen] = useState<boolean[]>([]);
+
+  useEffect(() => {
+    // Setting all notifications to be open by default
+    var listTrue = new Array(notifications.length).fill(true);
```
Comment (nichosystem) [inline apps/dapp/src/components/Notification/NotificationManager.tsx:23]: nit: let's stick to lets or consts for consistency

**Review by nichosystem — CHANGES_REQUESTED** [review]
just two nits to clean this up otherwise it looks good 👍

File: apps/dapp/src/components/Notification/NotificationManager.tsx (hunk 1 above)
Comment (MrMysteryMeat) [reply apps/dapp/src/components/Notification/NotificationManager.tsx]: done

File: apps/dapp/src/components/Notification/NotificationManager.tsx (hunk 2 above)
Comment (MrMysteryMeat) [reply apps/dapp/src/components/Notification/NotificationManager.tsx]: done

**Review by nichosystem — APPROVED** [review]

**Pull request merged**